debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -603,14 +604,20 @@ else if (data instanceof java.util.Date) {
     @Override
     protected Object convertTimeWithZone(Column column, Field fieldDefn, Object data) {
```

Comment:
@gunnarmorling @jpechane, can either of you offer your opinion of how I handle `timetz` consistency here?  This makes the `RecordsSnapshotProducerIT` and `RecordsStreamProducerIT` tests pass but I'm not absolutely certain this is the right approach.

Reply:
@Naros Is this stated inside a docs or standard? I've found that `java.sql.Time` accepts millis in its constructor. But it is non-final class so if there would be a subcalss of it that would provide also microseconds and nanoseconds then it is perfectly valid.
I've checked the code of the JDBC driver and they are already truncating the microseconds. The problem is that while snapshot provides milliseconds, streaming is providing microseconds.
So my opinion is
1) Do not truncate millis, always handle it as microsecond precision is coming
2) Create a Jira issue that would ensure that Time will be read from postgres in that way that microseconds are not lost (effectivelly reading it as strign and parsing locally)

Reply:
@jpechane That seems reasonable.  So what I'll do for now is to change the test input data from being supplied as `13:51:30.123+02:00` rather than `13:51:30.123789+0200` and remove the truncation logic for this issue and as you suggested open a follow-up on handling `.123789` precision properly there for this corner case.

Reply:
@Naros, which one is that follow-up issue you mention above?

Reply:
@gunnarmorling https://issues.jboss.org/browse/DBZ-1260